### DISCUSSION TITLE
Laravel Mongodb in serverless

### DISCUSSION THREAD

**Discussion by masterbater** [jenssegers/laravel-mongodb#3094] [Q&A] [answered]
How to handle connection in aws lambda, in nodejs I could manage connection by following this

https://www.mongodb.com/docs/atlas/manage-connections-aws-lambda/

**Comment by GromNaN** [comment, marked as answer]
I wrote a tutorial on how to connect to MongoDB Atlas from AWS Lambda, including using AWS credentials: https://www.mongodb.com/docs/php-library/current/tutorial/aws-lambda/#aws-credentials

**Reply by masterbater** [reply]
> I wrote a tutorial on how to connect to MongoDB Atlas from AWS Lambda, including using AWS credentials: https://www.mongodb.com/docs/php-library/current/tutorial/aws-lambda/#aws-credentials

Does the library handle the connections? or is it handled by bref?

**Reply by GromNaN** [reply]
Bref provides the PHP runtime with and mongodb extension.

To connect to Atlas, you may use a connection string provided by the Atlas UI:

```php

'connections' => [
    'mongodb' => [
        'driver' => 'mongodb',
        'dsn' => 'mongodb+srv://cluster0.example.mongodb.net/?authSource=%24external&authMechanism=MONGODB-AWS&retryWrites=true&w=majority',
        'database' => 'sample_mflix',
    ],
],


```

The credentials are provided to PHP by the lambda runtime with environment variables and read directly by the mongodb extension. You don't have to worry about passing the credentials.

When running the application outside of AWS, you need to provide traditional Atlas credentials.

**Reply by masterbater** [reply]
> Bref provides the PHP runtime with and mongodb extension.
> 
> 
> 
> To connect to Atlas, you may use a connection string provided by the Atlas UI:
> 
> 
> 
> ```php
> 
> 
> 
> 'connections' => [
> 
>     'mongodb' => [
> 
>         'driver' => 'mongodb',
> 
>         'dsn' => 'mongodb+srv://cluster0.example.mongodb.net/?authSource=%24external&authMechanism=MONGODB-AWS&retryWrites=true&w=majority',
> 
>         'database' => 'sample_mflix',
> 
>     ],
> 
> ],
> 
> 
> 
> 
> 
> ```
My point is managing the connection outside the handler, in nodejs it is advisable to put it outside so that connection will be persistent or else it will always create a new connection, in high concurrency workload atlas connection limitation will be easily reach the threshold

**Reply by GromNaN** [reply]
Whatever the environment, PHP will initialize a connection at the start of the request and close it at the end.
Unless you are using a "worker" mode. But be careful, because if the lambda is frozen for too long the connection will be lost. I never tested with Laravel.

**Reply by masterbater** [reply]
> Whatever the environment, PHP will initialize a connection at the start of the request and close it at the end. Unless you are using a "worker" mode. But be careful, because if the lambda is frozen for too long the connection will be lost. I never tested with Laravel.

Thanks so if there is 1000 concurrent then php will open 1000 connection and close it.  I think Ill just keep using fargate

**Reply by GromNaN** [reply]
I don't think using Fargate or any other environment makes any difference. It's the mongodb PHP extension that handles the connection.

**Reply by masterbater** [reply]
> I don't think using Fargate or any other environment makes any difference. It's the mongodb PHP extension that handles the connection.

So you mean mongodb php doesnt maintain a connection like mongodb nodejs pooling

https://www.mongodb.com/docs/drivers/node/current/faq/#how-does-connection-pooling-work-in-the-node-driver-
